core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -635,6 +635,96 @@ public UserCancelledException(String message, Throwable cause, String recoverySu
         }
     }
 
+    /**
+     * Could not perform the operation due to incorrect lambda configuration.
+     */
+    public static class UnexpectedLambdaException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid lambda response error.";
+        private static final String RECOVERY_SUGGESTION =
+                "Make sure that the lambda configuration is correct.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public UnexpectedLambdaException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the operation due to incorrect lambda configuration.
+     */
+    public static class UserLambdaValidationException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid lambda response error.";
+        private static final String RECOVERY_SUGGESTION =
+                "Make sure that the lambda configuration is correct.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public UserLambdaValidationException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the operation due to invalid sms role trust relationship.
+     */
+    public static class InvalidSmsRoleTrustRelationshipException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid SMS Role Trust Relationship error.";
+        private static final String RECOVERY_SUGGESTION =
```

Comment:
I believe you have the same class in PR 1884 ?

Reply:
this branch is based off the updateuserattribute branch. The updateuserattribute branch needs to be merged first

Reply:
You can change the base to 1884's branch and when 1884 is merged GH will automatically update this PR's base as dev-preview.